src/dlgProfilePreferences.cpp

```diff
@@ -703,6 +718,84 @@ void dlgProfilePreferences::initWithHost(Host* pHost)
     }
 
     timeEdit_timerDebugOutputMinimumInterval->setTime(pHost->mTimerDebugOutputSuppressionInterval);
+    bool socketErrorHandled = false;
+    notificationArea->hide();
+    notificationAreaIconLabelWarning->hide();
+    notificationAreaIconLabelError->hide();
+    notificationAreaIconLabelInformation->hide();
+    notificationAreaMessageBox->hide();
+
+#if !defined(QT_NO_SSL)
+    if (QSslSocket::supportsSsl() && pHost->mSslTsl == true) {
+        QSslCertificate cert = pHost->mTelnet.getPeerCertificate();
+        ssl_issuer_label->setText(cert.issuerInfo(QSslCertificate::CommonName).join(","));
+        ssl_issued_label->setText(cert.subjectInfo(QSslCertificate::CommonName).join(","));
+        ssl_expires_label->setText(cert.expiryDate().toString(Qt::LocalDate));
+        ssl_serial_label->setText(QString::fromStdString(cert.serialNumber().toStdString()));
+        checkBox_self_signed->setStyleSheet("");
+        checkBox_expired->setStyleSheet("");
+        ssl_issuer_label->setStyleSheet("");
+        ssl_expires_label->setStyleSheet("");
+        checkBox_ssl->setStyleSheet("");
+
+        if (!pHost->mTelnet.getSslErrors().empty()) {
```

Comment:
Could you clear the errors when the dialog is closed? Right now if there's an error and you change a preference, close the dialog, connect successfully, and open it again - the error is still there.